### DISCUSSION TITLE
Listing files

### DISCUSSION THREAD

**Discussion by lllama** [koltyakov/gosip#77] [Q&A]
Would it be possible to get an example of how to list files in a folder? I'm trying to list and download all the files in the "Shared Documents" folder on my site. Bonus points for decending into folders.

Thanks for the great library!

**Comment by koltyakov** [comment]
Hi @lllama, thank you!

Did you have a chance taking a look into the docs?

https://go.spflow.com/samples/documents

Sounds that it contains what you asking:

```go
// Getting folder object
foler := sp.Web().GetFolder("MyLibrary/Folder01")

// Getting folder's files (entry object)
files := foler.Files()
```

and [downloading a file](https://go.spflow.com/samples/documents#downloading-files).

**Comment by lllama** [comment]
Thanks @koltyakov! I was just missing a couple of steps to get the filenames for the folder. I came up with the following:

```golang
files := sp.Web().GetFolder("Shared%20Documents").Files()
itemsResp, _ := files.Get()

for _, file := range itemsResp.Data() {
    fmt.Println(file.Data().Name)
}
```
